### DISCUSSION TITLE
Azure platform - missing file fetch silently fails

### DISCUSSION THREAD

**Discussion by ondrej-langr** [renovatebot/renovate#26474] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

Hello, i stumbled upon this randomly: When config is being fetched locally (azure devops - /lib/modules/platform/azure/index.ts line 149 on main) and the requested file does not exist the actual response from azure git is this:
```json
{
  "$id": "1",
  "innerException": null,
  "message": "TF401174: The item 'something.js' could not be found in the repository '<redacted>' at the version specified by '<Branch: main >' (resolved to commit '<redacted>')",
  "typeName": "Microsoft.TeamFoundation.Git.Server.GitItemNotFoundException, Microsoft.TeamFoundation.Git.Server",
  "typeKey": "GitItemNotFoundException",
  "errorCode": 0,
  "eventId": 3000
}
```
which does not throw! This results in renovate having "empty" config and continues its journey without exiting or even saying logging about this.

Im open to contributing, but I wanted to discuss this first before resolving this here or continuing this to 'azure-devops-node-api' (or maybe this is intended? Not sure)

FYI: I have a script to reproduce, but it should be apparent whats wrong ;)

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
PR would be welcomed to catch this result and treat it like a 404

**Reply by ondrej-langr** [reply]
I have something in mind already - simple .includes() would suffice imho. I will create a PR today or tommorow and let you know
